rfd/0109-cloud-agent-upgrades.md

```diff
@@ -327,6 +327,30 @@ to be changed to follow the same pattern of registering a release channel and in
 by ec2 discovery.
 
 
+### ECS Fargate Agents
+
+When a user uses the Teleport Discover flow to connect an RDS, a Teleport agent is automatically provisioned using ECS Fargate. We need a
+way to keep these agents updated. One idea is to run a separate goroutine in Teleport that periodically checks for an available update and
+uses the ECS Fargate API to update Teleport agents.
+
+This goroutine does not need to be run on all Teleport clusters. It will only be needed if the `TELEPORT_AUTOMATIC_UPGRADES` environment
+variable is enabled. The window when Fargate agents are updated should also coincide with the tenant upgrade window.
+
+The logic flow might look something like:
+- List all available Fargate agents
+- If no agents are available, do nothing
+- If agents are available, query the current cloud-stable agent version
+- Update all Fargate agents that do not match the current cloud-stable agent version
+
+The goroutine will also need to handle critical updates. Periodically, the goroutine will check the critical version endpoint if a critical
```

Comment:
I think this paragraph about critical updates should actually be just included into logic flow description. It probably should be very similar to the `core upgrade model` diagram shown above -
1. We check if critical update is available
2. If yes, list all agents and try upgrade them if they are lower version, regardless of maintenance window 
3. If no we check if we're currently inside maintenance window, if no, do nothing
4. If we're in maintenance window list agents and try to upgrade those which need it.

We also will only need to list agents if we're in a maintenance or there's a critical update.
The diagram also mentions aspect of agent healthiness, I'm not sure how it translates into Fargate or if it's applicable, but probably something to consider.

Reply:
I've updated the flow to match the core upgrade model. They should be pretty much the same.